Handle https radar links in bug import script

pascoej opened the pull request:
#### 7b66f5cd754e147253db994967b234cea0aa060c
<pre>
Handle https radar links in bug import script
<a href="https://bugs.webkit.org/show_bug.cgi?id=273414">https://bugs.webkit.org/show_bug.cgi?id=273414</a>
<a href="https://rdar.apple.com/127240471">rdar://127240471</a>

Reviewed by Jonathan Bedard.

Whenever you copy and paste a radar link from the GitHub PR page, a version of the
url that is of the form <a href="https://rdar.apple.com/problem/127122951">https://rdar.apple.com/problem/127122951</a> gets copied. This
format doesn&apos;t seem to get recognized by our bug importer script.

This patch fixes that by adding regex for that format.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py:
(Tracker):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/branch_unittest.py:
(TestBranch.test_manual_radar_cc_integer):
(TestBranch):
(TestBranch.test_manual_radar_cc_http):

Canonical link: <a href="https://commits.webkit.org/278175@main">https://commits.webkit.org/278175@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/80e1962cd3b53c220472539ec9ece2ca2cb1afe5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/49565 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/28850 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/52612 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/52807 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/241 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/51869 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/34870 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/26470 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/40468 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/51665 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/26385 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/42701 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/21583 "Passed tests") | 
| [✅ 🧪 webkitpy](https://ews-build.webkit.org/#/builders/6/builds/49421 "Passed tests") | [⏳ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/iOS-17-Simulator-WPT-WK2-Tests-EWS "Waiting to run tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/43884 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/7936 "Built successfully") | 
| | [⏳ 🧪 api-ios](https://ews-build.webkit.org/#/builders/API-Tests-iOS-Simulator-EWS "Waiting to run tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/44390 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/54383 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/24650 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/20826 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/47838 "Passed tests") | 
| [✅ 🧪 services](https://ews-build.webkit.org/#/builders/28/builds/49591 "Passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/25919 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/8/builds/42846 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/46860 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10911 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/26763 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/25643 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

JonWBedard commented on Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/branch.py:
```diff
@@ -118,6 +118,8 @@ def cc_radar(cls, args, repository, issue):
                 sys.stdout.flush()
                 input = Terminal.input(': ')
+                if input.startswith("https://rdar.apple.com/"):
+                    input = input.removeprefix("https://rdar.apple.com/")
```
I think the correct place for this change is actually `Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py`, and just involves adding a new regex to `Tracker.RES`. I would expect that making the change there makes `test_manual_radar_cc_http` still pass

pascoej committed:
Handle https radar links in bug import script
https://bugs.webkit.org/show_bug.cgi?id=273414
rdar://127240471

Reviewed by Jonathan Bedard.

Whenever you copy and paste a radar link from the GitHub PR page, a version of the
url that is of the form https://rdar.apple.com/problem/127122951 gets copied. This
format doesn't seem to get recognized by our bug importer script.

This patch fixes that by adding regex for that format.

* Tools/Scripts/libraries/webkitbugspy/webkitbugspy/radar.py:
(Tracker):
* Tools/Scripts/libraries/webkitscmpy/webkitscmpy/test/branch_unittest.py:
(TestBranch.test_manual_radar_cc_integer):
(TestBranch):
(TestBranch.test_manual_radar_cc_http):

Canonical link: https://commits.webkit.org/278175@main
